### PR TITLE
Feature: set tags using the manifest

### PR DESCRIPTION
**Pull request by DeviaVir**
Because SpotFleets do not support settings tags, we use `spacel-agent` to set tags on runtime. We need tags to identify instances to deploy to (on `git-deploy`).

### CODE REVIEW

File: src/spacel/aws/tag.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import logging
+
+logger = logging.getLogger('spacel')
+
+
+class TagWriter():
```
Comment (thepwagner) [inline src/spacel/aws/tag.py]: Nit: `class TagWriter(object):`

https://wiki.python.org/moin/NewClassVsClassicClass

File: src/spacel/aws/tag.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    def update(self, manifest):
+        tags = manifest.tags
+        if not tags:
+            logger.debug('Tags not set.')
+            return
+
+        my_tags = []
+        for tag in tags:
+            prepared_tag = self._prepare(tag)
```
Comment (thepwagner) [inline src/spacel/aws/tag.py]: Could this use key/values in a dict?
`{"foo":"bar", "foz":"baz"}` is easier/cleaner than `[{"key":"foo","value":"bar"},{"key":"foz","value":"baz"}]

You could get away with list comprehension for the whole kit: `my_tags = [{'Key':k, 'Value': v} for k, v in tags.items() if k and v]`?

File: src/test/aws/test_tag.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+from spacel.aws.tag import TagWriter
+from test.aws import MockedClientTest, INSTANCE_ID
+
+TAG_KEY = 'key'
+TAG_VALUE = 'value'
+
+
+class TestTagWriter(MockedClientTest):
+    def setUp(self):
+        super(TestTagWriter, self).setUp()
```
Comment (thepwagner) [inline src/test/aws/test_tag.py]: Dat code re-use!

**Comment by thepwagner** [comment]
+1 on impl, just pondering API

File: src/spacel/aws/tag.py (hunk 2 above)
Comment (DeviaVir) [inline src/spacel/aws/tag.py]: :+1: yup, we do not _have_ to keep annoying API's around.

**Comment by thepwagner** [comment]
+1, purdy

**Pull request merged**